api/preProcessor-template.ts

```diff
@@ -0,0 +1,37 @@
+import thing from 'module';
+
+export let sourceExtensionTypes: string[] = [".ts", ".tsx"];
+
+export let outputExtensionTypes: string[] = [".mjs", ".cjs"];
+
+
+// Example of defining "options" for a TypeScript loader
+export interface TypeScriptOptions {
+  tsconfigFile?: string
+}
+
+interface ProcessedFile {
+  source: string,
+  extension?: string
+};
+
+interface preProcessorInstance {
+  process: Function
+};
+
+export function getPreprocessor(options: TypeScriptOptions = {}): preProcessorInstance {
+  // Create instances and state variables
+  // E.g. a CompilerHost instance to compile TypeScript 
+
+
+  // Example of returning a transpiled source using options provided by project config
+  return { 
+          process(source: string): ProcessedFile{
```

Comment:
The spacing here looks suspicious - can you check if it would be fixed by prettier?

Reply:
Good called, fixed! Although will have to be further tweaked now that it will return a `module` update coming.